homeassistant/components/climate/xiaomi_miio.py

```diff
@@ -0,0 +1,624 @@
+        self.hass = hass
+        self._name = name if name else DEFAULT_NAME
+        self._target_humidity = target_humidity
+        self._away = away
+        self._hold = hold
+        self.host = host
+        self.token = token
+        self.sync = sync
+        self._customize = customize
+
+        self._climate = None
+        self._state = None
+        self._state = self.climate_get_state()
+
+        self._target_temperature = self._state.temp
+        self._current_operation = self._state.operation
+
+        self._current_humidity = current_humidity
+        self._aux = aux
+        self._operation_list = ['heat', 'cool', 'auto', 'off']
+
+        if self._customize and ('fan' in self._customize):
+            self._customize_fan_list = list(self._customize['fan'])
+            self._fan_list = self._customize_fan_list
+            self._current_fan_mode = 'idle'
+        else:
+            self._fan_list = ['low', 'medium', 'high', 'auto']
+            self._current_fan_mode = self._state.wind_force
+
+        if self._customize and ('swing' in self._customize):
+            self._customize_swing_list = list(self._customize['swing'])
+            self._swing_list = self._customize_swing_list
+            self._current_swing_mode = 'idle'
+        else:
+            self._swing_list = ['on', 'off']
+            self._current_swing_mode = self._state.sweep
+        self._target_temperature_high = target_temp_high
+        self._target_temperature_low = target_temp_low
+        self._max_temp = target_temp_high + 1
+        self._min_temp = target_temp_low - 1
+        self._target_temp_step = DEFAULT_STEP
+
+        self._unit_of_measurement = TEMP_CELSIUS
+        self._current_temperature = None
+        self._sensor_entity_id = sensor_entity_id
+
+        if sensor_entity_id:
+            async_track_state_change(
+                hass, sensor_entity_id, self._async_sensor_changed)
+            sensor_state = hass.states.get(sensor_entity_id)
+            if sensor_state:
+                self._async_update_temp(sensor_state)
+        if sync:
+            async_track_time_interval(
+                hass, self._async_get_states, timedelta(seconds=self.sync))
+
+    @callback
+    def _async_update_temp(self, state):
+        """Update thermostat with latest state from sensor."""
+        unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+
+        try:
+            self._current_temperature = self.hass.config.units.temperature(
+                float(state.state), unit)
+        except ValueError as ex:
+            _LOGGER.error('Unable to update from sensor: %s', ex)
+
+    @asyncio.coroutine
+    def _async_sensor_changed(self, entity_id, old_state, new_state):
+        """Handle temperature changes."""
+        if new_state is None:
+            return
+        self._async_update_temp(new_state)
+        # yield from self.async_update_ha_state()
+
+    @asyncio.coroutine
+    def _async_get_states(self, now=None):
```

Comment:
This should be the `async_update` method that is being polled if `should_poll` is `True`.